src/inmanta/deploy/tasks.py

```diff
@@ -45,13 +46,17 @@ class Task(abc.ABC):
     """
 
     resource: ResourceIdStr
-
     id: resources.Id = dataclasses.field(init=False, compare=False, hash=False)
+    reason: typing.Optional[str] = dataclasses.field(compare=False, hash=False, default=None)
```

Comment:
Why is this field optional? In which situations is this None?

Reply:
I have added a comment, let me know if it is better